src/Stan/Inspection/AntiPattern.hs

```diff
@@ -388,3 +390,15 @@ stan0215 = mkAntiPatternInspection (Id "STAN-0215") "Slashes in paths" (FindAst
     pathLit :: PatternAst
     pathLit = PatternAstConstant (ContainStr "/")
         ||| PatternAstConstant (ContainStr "\\\\")
+
+-- | 'Inspection' - use 'Text' or 'ByteString'  instead of 'String' @STAN-0216@.
+stan0216 :: Inspection
+stan0216 = mkAntiPatternInspection (Id "STAN-0216") "String"
+  (FindAst $ PatternAstName strNameMeta (?))
```

Comment:
`PatternAstName strNameMeta (?)` is not exactly what is needed in here. This one would look for any `String` in the code.
In this case, we need to check that the `String` is in type, specifically, that it is the returned value. 

Maybe this example can be of use:
https://github.com/kowainik/stan/blob/43dd2c7cd27f7a5f729586d551438df9e5efa4f8/src/Stan/Inspection/Style.hs#L54-L67

In this example, we search for big tuples. For `String` we would need to do something similar as well. 

If that doesn't make sense, just ping us with any questions 🙂

Reply:
You mean to do something like this?
```haskell
stan0216 :: Inspection
stan0216 = Inspection
    { inspectionId = Id "STAN-0216"
    , inspectionName = "String"
    , inspectionDescription = "Usage of the default 'String' Haskell type"
    , inspectionSolution =
        [ "Consider using instead 'Text' for readable text, or 'ByteString' otherwise"
        ]
    , inspectionCategory = Category.antiPattern :| [Category.syntax]
    , inspectionSeverity = Style
    , inspectionAnalysis = StringUsage
    }
```
It's not really evident to me how to implement this `StringUsage` matcher, sorry 😖

Reply:
@kutyel Yeah, you need to introduce a separate constructor for the `InspectionAnalysis` type. `StringUsage` works fine :+1: 

You can see the implementation of `analyseBigTuples` to check how such custom check is implemented:

https://github.com/kowainik/stan/blob/43dd2c7cd27f7a5f729586d551438df9e5efa4f8/src/Stan/Analysis/Analyser.hs#L69-L84

Reply:
I arrived to the point in which I had to compare an `HieAST TypeIndex` to a "String", how can I do that? 🤔

Reply:
@kutyel You need to implement a pattern to match the `String` type. I guess the most reliable way to figure out, how the pattern should look like, is to debug the content of the HIE files and see what's inside. For this you need to write a function you want to view and follow the instructions from our wiki on how to debug HIE files in GHCi:

* https://github.com/kowainik/stan/wiki/Development#view-hie-ast-of-a-file